### PR TITLE
Plugin: Bump minimum required WordPress version to 6.6

### PR DESCRIPTION
**Pull request by ockham**
## What?
Similar to #64126.

Bumps the "Required WP version" field to 6.6, and the "Tested up to" field to 6.7. It also updates the performance test base hash.

## Why?
To reflect the current WP version, and update the minimum required version to the previous one.

I'd also like to unblock https://github.com/WordPress/gutenberg/pull/64676 😬 

## How?
See https://developer.wordpress.org/block-editor/explanations/architecture/performance/#update-the-reference-commit

## Testing Instructions
* CI checks should be green.
* Confirm that new commit SHA matches guidelines.

### CODE REVIEW

**Comment by ntsekouras** [comment]
Shouldn't we remove first (or at the same PR) the compat layer for 6.6?

**Comment by ockham** [comment]
> Shouldn't we remove first (or at the same PR) the compat layer for 6.6?

Sure! I see that's what you did for 6.5 (https://github.com/WordPress/gutenberg/pull/64096 before https://github.com/WordPress/gutenberg/pull/64126). Looks like you also had to remove the 6.4 compat layer in #64096 since I didn't do that in #60780 🙈 So it's probably a good idea to do it at once.

I'll push a commit 👍

File: lib/rest-api.php [hunk 1]
```diff
@@ -19,11 +19,10 @@
  * @return array Array of arguments for registering a post type.
  */
 function gutenberg_override_global_styles_endpoint( array $args ): array {
-	$args['rest_controller_class']           = 'WP_REST_Global_Styles_Controller_Gutenberg';
-	$args['revisions_rest_controller_class'] = 'Gutenberg_REST_Global_Styles_Revisions_Controller_6_6';
```
Comment (ockham) [inline lib/rest-api.php:23]: @Mamaduka Am I doing this right? 😄 (Reference: https://github.com/WordPress/gutenberg/pull/65259)

Comment (ramonjd) [reply lib/rest-api.php:23]: This looks good to me. 

The properties were added to support 6.5, but 6.6 specifies controllers in the post type registration now:

https://github.com/WordPress/wordpress-develop/blob/6.6/src/wp-includes/post.php#L488

**Comment by ramonjd** [comment]
Thanks @ockham I smoke tested the site editor and the revisions controller stuff in particular. From my perspective LGTM

File: lib/rest-api.php (hunk 1 above)
Comment (Mamaduka) [reply lib/rest-api.php:23]: Looks correct to me as well 👍

**Review by Mamaduka — APPROVED** [review]
Smoke-tested editors and everything looks to be working correctly.

No PHP errors or warnings in logs.

File: lib/load.php [hunk 2]
```diff
@@ -70,18 +65,6 @@ function gutenberg_is_experiment_enabled( $name ) {
 require __DIR__ . '/compat/plugin/edit-site-routes-backwards-compat.php';
 require __DIR__ . '/compat/plugin/fonts.php';
 
-// The Token Map was created during 6.6 in order to support the HTML API. It must be loaded before it.
-require __DIR__ . '/compat/wordpress-6.6/class-gutenberg-token-map-6-6.php';
-require __DIR__ . '/compat/wordpress-6.7/class-gutenberg-token-map-6-7.php';
```
Comment (SantosGuillamot) [inline lib/load.php:75]: This is removing a require from a file in `6.7` compat folder. Do you know if that's expected?

Comment (Mamaduka) [reply lib/load.php:75]: Good catch. The WP 6.7 files shouldn't be removed here.

Comment (ockham) [reply lib/load.php:75]: Ah, good catch indeed! Oversight on my part, I'll add it back 😄

Comment (ockham) [reply lib/load.php:75]: f602432a615

**Comment by desrosj** [comment]
Just a heads up that when we bump minimum version requirements, we do need to make a Core SVN change to pin the last compatible version of the Gutenberg plugin in the E2E test workflow for any branch losing support. I've gone and opened #67167 to add this to the documentation, and [Core-62488](https://core.trac.wordpress.org/ticket/62488) to make this change once 19.8 enters RC phase.

**Comment by ockham** [comment]
TIL 😅 Thank you @desrosj!

**Comment by ockham** [comment]
> Just a heads up that when we bump minimum version requirements, we do need to make a Core SVN change to pin the last compatible version of the Gutenberg plugin in the E2E test workflow for any branch losing support. I've gone and opened #67167 to add this to the documentation, and [Core-62488](https://core.trac.wordpress.org/ticket/62488) to make this change once 19.8 enters RC phase.

PR here: https://github.com/WordPress/wordpress-develop/pull/8008

**Pull request merged**